crates/terminal/src/terminal.rs

```diff
@@ -1727,20 +1725,26 @@ impl Terminal {
         self.task.as_ref()
     }
 
-    pub fn wait_for_completed_task(&self, cx: &App) -> Task<()> {
-        if let Some(task) = self.task() {
-            if task.status == TaskStatus::Running {
-                let completion_receiver = task.completion_rx.clone();
-                return cx.spawn(|_| async move {
-                    let _ = completion_receiver.recv().await;
-                });
+    pub fn wait_for_completed_task(&self, cx: &App) -> Task<i32> {
```

Comment:
Return value of this function should probably be `Result<Task<i32>>`, as right now we're intermixing not having a task to poll or a task with unknown exit code with a legit return value.